americangut/diversity_analysis.py

```diff
@@ -0,0 +1,1729 @@
+
+        dist_bar = np.array(dist_bar)
+        dist_std = np.array(dist_std)
+
+        # Creates the boxplot
+        ax, xpos = barchart(dist_bar,
+                            ax=ax,
+                            errors=dist_std,
+                            interval=interval,
+                            width=width,
+                            offset=offset,
+                            xticklabels=order,
+                            colormap=colormap,
+                            match_colors=match_colors,
+                            elinewidth=elinewidth,
+                            ecapwidth=ecapwidth)
+        # Watches the xlimits
+        if id1 == 0:
+            xlim = [xpos[0] - interval*0.75]
+
+        # Gets the critical lines to display on the figure
+        if lowest is None:
+            lowest, __ = _get_bar_height(dist_bar+dist_std)
+        if p_values is not None and (ref_loc == 0 or len(dist_bar) == 2):
+            bars = add_comparison_bars(xpos,
+                                       dist_bar+dist_std,
+                                       p_values,
+                                       ax,
+                                       lowest=lowest,
+                                       label_size=label_size,
+                                       show_value=show_p_value)
+        elif p_values is not None and ref_loc == (len(dist_bar) - 1):
+            bars = add_comparison_bars(xpos[::-1],
+                                       (dist_bar+dist_std)[::-1],
+                                       p_values[::-1],
+                                       ax,
+                                       lowest=lowest,
+                                       label_size=label_size,
+                                       show_value=show_p_value)
+        elif p_values is not None:
+            bars = add_comparison_bars(xpos[ref_loc:],
+                                       dist_bar[ref_loc:] + dist_std[ref_loc:],
+                                       p_values[(ref_loc):],
+                                       ax, lowest=lowest,
+                                       label_size=label_size,
+                                       show_value=show_p_value)
+            bars2 = add_comparison_bars(xpos[:(ref_loc+1)][::-1],
+                                        dist_bar[:(ref_loc+1)][::-1] +
+                                        dist_std[:(ref_loc+1)][::-1],
+                                        p_values[:(ref_loc)][::-1],
+                                        ax, lowest=lowest,
+                                        label_size=label_size,
+                                        show_value=show_p_value)
+            bars.extend(bars2)
+
+        # Advances the bar count
+        bar_counter = bar_counter + 1
+    xlim.append(xpos[-1] + interval*0.75)
+
+    # Sets up axis formatting defaults, if not supplied
+    if 'show_frame' not in kwargs:
```

Comment:
Just a note, and this is minor and non-blocking, but you can reduce the lines of code (if you haven't seen it, commonly referred to as LOC), by doing something like:

```python
kwargs['show_frame'] = kwargs.get('show_frame', False) 
```

LOC comes up in formal measures of code complexity. The basic principle is that, more lines of code, means more for the developer to digest increasing the difficulty of fully understanding a method. Its somewhat analogous to a run-on sentence vs. a short succinct one where the latter is generally easier and quicker for the reader to understand.